tests/e2e/cases/02_ls.sh

```
@@ -0,0 +1,11 @@
+#!/usr/bin/env bash
+# pvm ls discovers the installed version with [cli, fpm] tags.
+set -euo pipefail
+source "$(dirname "$0")/../_lib.sh"
+
+step "pvm ls — version discovery + package tags"
+"$PVM_BIN" ls
+"$PVM_BIN" ls | grep -q "$PREVIOUS" || fail "pvm ls missing $PREVIOUS"
+"$PVM_BIN" ls | grep -E "$PREVIOUS.*cli.*fpm" >/dev/null \
+    && ok "pvm ls shows [cli, fpm] for $PREVIOUS" \
```

Shellcheck (0.11.0)

[info] Note that A && B || C is not if-then-else. C may run when A is true.

(SC2015)